experimental/unitsconversion/src/measureunit.rs

```diff
@@ -0,0 +1,269 @@
+            (60, part)
+        } else if let Some(part) = part.strip_prefix("zebi") {
+            (70, part)
+        } else if let Some(part) = part.strip_prefix("yobi") {
+            (80, part)
+        } else {
+            (0, part)
+        }
+    }
+
+    // TODO: consider using a sufficient trie search for finding the unit id.
+    /// Get the unit id.
+    /// NOTE:
+    ///    if the unit id is found, the function will return (unit id, part without the unit id and without `-` at the beginning of the remaining part if it exists).
+    ///    if the unit id is not found, the function will return None.
+    fn get_unit_id<'data>(
+        part: &'data str,
+        trie: &ZeroTrie<ZeroVec<'data, u8>>,
+    ) -> Option<(usize, &'data str)> {
+        // TODO(#4379): this is inefficient way to search for an item in a trie.
+        // we must implement a way to search for a prefix in a trie.
+        let mut result = None;
+        for (index, _) in part.char_indices() {
+            let identifier = &part[..=index];
+            if let Some(value) = trie.get(identifier.as_bytes()) {
+                result = Some((value, &part[identifier.len()..]));
+            }
+        }
+
+        result
+    }
+
+    /// Process a part of an identifier.
+    /// For example, if the whole identifier is: "square-kilometer-per-second",
+    /// this function will be called for "square-kilometer" with sign (1) and "second" with sign (-1).
+    fn analyze_identifier_part(
+        identifier_part: &str,
+        sign: i8,
+        result: &mut Vec<MeasureUnitItem>,
+        trie: &ZeroTrie<ZeroVec<'_, u8>>,
+    ) -> Result<(), ConversionError> {
+        let mut identifier = identifier_part;
+        while !identifier.is_empty() {
+            let (power, identifier_after_power) = Self::get_power(identifier);
+            let (si_prefix, identifier_after_si) = Self::get_si_prefix(identifier_after_power);
+            let (unit_id, identifier_after_unit) =
+                match Self::get_unit_id(identifier_after_si, trie) {
+                    Some((unit_id, identifier_after_unit)) => (unit_id, identifier_after_unit),
+                    None => return Err(ConversionError::InvalidUnit),
+                };
+
+            result.push(MeasureUnitItem {
+                power: power as i8 * sign,
+                si_prefix,
+                unit_id: unit_id as u16,
+            });
+
+            identifier = identifier_after_unit.get(1..).unwrap_or("");
+        }
+
+        Ok(())
+    }
+
+    // TODO: add test cases for this function.
+    /// Process an identifier.
+    pub fn try_from_identifier<'data>(
+        identifier: &'data str,
+        trie: &ZeroTrie<ZeroVec<'data, u8>>,
+    ) -> Result<Vec<MeasureUnitItem>, ConversionError> {
+        let (num_part, den_part) = identifier
+            .split_once("-per-")
+            .or_else(|| identifier.split_once("per-"))
+            .unwrap_or((identifier, ""));
```

Comment:
I realized we actually only need one search:

```suggestion
        let (num_part, den_part) = identifier
            .split_once("per-")
            .map(|(n,d)| (n.strip_suffix("-").unwrap_or(n), d))
            .unwrap_or((identifier, ""));
```

Reply:
okay, done, but I feel we over complicated it